lib/smart_proxy_ansible/roles_reader.rb

```diff
@@ -34,34 +34,27 @@ def logger
         private
 
         def read_roles(roles_path)
-          rescue_and_raise_file_exception ReadRolesException,
-                                          roles_path, 'roles' do
-            glob_path("#{roles_path}/*").map do |path|
-              path.split('/').last
-            end
+          glob_path("#{roles_path}/*").map do |path|
+            path.split('/').last
           end
+        rescue Errno::ENOENT, Errno::EACCES => e
+          logger.debug(e.backtrace)
+          message = "Could not read Ansible roles #{roles_path} - #{e.message}"
+          raise ReadRolesException.new(message), message
         end
 
         def glob_path(path)
           Dir.glob path
         end
 
         def roles_path_from_config
-          rescue_and_raise_file_exception ReadConfigFileException,
-                                          DEFAULT_CONFIG_FILE, 'config file' do
-            File.readlines(DEFAULT_CONFIG_FILE).select do |line|
-              line =~ /^\s*roles_path/
-            end
+          File.readlines(DEFAULT_CONFIG_FILE).select do |line|
+            line =~ /^\s*roles_path/
           end
-        end
-
-        def rescue_and_raise_file_exception(exception, path, type)
-          yield
         rescue Errno::ENOENT, Errno::EACCES => e
           logger.debug(e.backtrace)
-          exception_message = "Could not read Ansible #{type} "\
-            "#{path} - #{e.message}"
-          raise exception.new(exception_message), exception_message
+          message = "Could not read Ansible config file #{DEFAULT_CONFIG_FILE} - #{e.message}"
+          raise ReadRolesException.new(message), message
```

Comment:
I would expect tests to pass when you change it to `raise ReadConfigFileException`.

Reply:
I thought I had that but looks like it got lost somewhere. Updated.